examples/api/lib/material/selection_area/custom_container.dart

```diff
@@ -0,0 +1,136 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// This sample demonstrates how to create a [SelectionContainer] that only
+// allows selecting everything or nothing with no partial selection.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+
+void main() => runApp(const MyApp());
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+
+  static const String _title = 'Flutter Code Sample';
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: _title,
+      home: SelectionArea(
+        child: Scaffold(
+          appBar: AppBar(title: const Text(_title)),
+          body: Center(
+            child: SelectionAllOrNoneContainer(
+              child: Column(
+                mainAxisAlignment: MainAxisAlignment.center,
+                children: const <Widget>[
+                  Text('Row 1'),
+                  Text('Row 2'),
+                  Text('Row 3'),
+                ],
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class SelectionAllOrNoneContainer extends StatefulWidget {
+  const SelectionAllOrNoneContainer({
+    super.key,
+    required this.child
+  });
+
+  final Widget child;
+
+  @override
+  State<StatefulWidget> createState() => _SelectionAllOrNoneContainerState();
+}
+
+class _SelectionAllOrNoneContainerState extends State<SelectionAllOrNoneContainer> {
+  final MySelectAllOrNoneContainerDelegate delegate = MySelectAllOrNoneContainerDelegate();
+
+  @override
+  void dispose() {
+    delegate.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final SelectionRegistrar? registrar = SelectionRegistrarScope.maybeOf(context);
```

Comment:
Is this still needed? I thought the SelectionContainer does this automatically now.. could simplify the example a little bit.